### PR TITLE
Improvement fu user implementation

### PR DESCRIPTION
**Pull request by srdjastankovic**
- Implement FU verification in connector
 - Implement ver on user space based on txt file
 - update typos

### CODE REVIEW

File: sources/firmware_update.c [hunk 1]
```diff
@@ -158,94 +177,153 @@ static void _handle_abort(firmware_update_t* firmware_update)
...
+    firmware_update->status = status;
 }
 
-void firmware_update_handle_parameter(firmware_update_t* firmware_update, firmware_update_t* parameter)
+static void _listener_on_firmware_update_status(firmware_update_t* firmware_update)
 {
     /* Sanity Check */
     WOLK_ASSERT(firmware_update);
-    WOLK_ASSERT(parameter);
+    WOLK_ASSERT(status);
```
Comment (LazarBozic94) [inline sources/firmware_update.c:202]: what is status?

File: sources/firmware_update.c [hunk 2]
```diff
@@ -158,94 +177,153 @@ static void _handle_abort(firmware_update_t* firmware_update)
...
+void firmware_update_init(firmware_update_t* firmware_update, firmware_update_start_installation_t start_installation,
+                          firmware_update_is_installation_completed_t is_installation_completed,
+                          firmware_update_verification_store_t verification_store,
+                          firmware_update_verification_read_t verification_read,
+                          firmware_update_get_version_t get_version, firmware_update_abort_t abort_installation,
+                          void* wolk_ctx)
 {
-    /* Sanity Check */
+    /* Sanity check */
     WOLK_ASSERT(firmware_update);
```
Comment (LazarBozic94) [inline sources/firmware_update.c:255]: assert wolk_ctx as well?

**Review by LazarBozic94 — CHANGES_REQUESTED** [review]

File: examples/full_feature_set/firmware_update_implementation.c [hunk 3]
```diff
@@ -13,44 +13,236 @@
...
+                 FIRMWARE_UPDATE_VERSION_FORMAT_DELIMITER, version_minor, FIRMWARE_UPDATE_VERSION_FORMAT_DELIMITER,
+                 version_patch)
         > FIRMWARE_UPDATE_VERSION_SIZE) {
         return false;
     }
 
     return true;
 }
 
+static bool _set_version(char* source, int8_t* version)
```
[the remark this answers is not in the document]
Comment (srdjastankovic) [reply examples/full_feature_set/firmware_update_implementation.c:56]: > Why do you start naming with an underscore?

inherit from wolk_connect.c, firmware_update.c

File: examples/full_feature_set/firmware_update_implementation.c [hunk 4]
```diff
@@ -13,44 +13,236 @@
...
+    version_string = strtok(NULL, FIRMWARE_UPDATE_VERSION_FORMAT_DELIMITER);
+
+    if (version_string == NULL) {
+        return false;
+    }
+    *version = atoi(version_string);
+    while (version_string != NULL) {
+        version_string = strtok(NULL, FIRMWARE_UPDATE_VERSION_FORMAT_DELIMITER);
+        if (version_string != NULL) {
+            version++;
```
[the remark this answers is not in the document]
Comment (srdjastankovic) [reply examples/full_feature_set/firmware_update_implementation.c:69]: not suported format by this imeplemntation, see this:
#define FIRMWARE_UPDATE_VERSION_NUMBER_OF_ELEMENTS 3
static int8_t version_major = 0;
static int8_t version_minor = 0;
static int8_t version_patch = 1;

File: examples/full_feature_set/firmware_update_implementation.c [hunk 5]
```diff
@@ -13,44 +13,236 @@
...
+
+    if (tmp_status != NULL) {
+        if (snprintf(file_content, FIRMWARE_UPDATE_VERSION_SIZE, "%d", tmp_status) > FIRMWARE_UPDATE_VERSION_SIZE) {
+            return false;
+        }
+    }
+    if (tmp_version != NULL) {
+        if (snprintf(file_content + strlen(file_content), FIRMWARE_UPDATE_VERSION_SIZE, "%s%d%s%d%s%d",
+                     FIRMWARE_UPDATE_VERIFICATION_FILE_DELIMITER, tmp_version[0],
+                     FIRMWARE_UPDATE_VERSION_FORMAT_DELIMITER, tmp_version[1], FIRMWARE_UPDATE_VERSION_FORMAT_DELIMITER,
```
[the remark this answers is not in the document]
Comment (srdjastankovic) [reply examples/full_feature_set/firmware_update_implementation.c:129]: clang?

File: examples/full_feature_set/firmware_update_implementation.c [hunk 6]
```diff
@@ -13,44 +13,236 @@
...
+}
+
+static bool _file_read(uint8_t* status, int8_t* version)
+{
+    char file_content[FIRMWARE_UPDATE_VERSION_SIZE];
+    memset(file_content, '\0', FIRMWARE_UPDATE_VERSION_SIZE);
+
+    FILE* file_pointer = fopen(FIRMWARE_UPDATE_VERIFICATION_FILE_NAME, FIRMWARE_UPDATE_VERIFICATION_FILE_MODE_READ);
+    if (file_pointer == NULL) {
+        file_pointer = fopen(FIRMWARE_UPDATE_VERIFICATION_FILE_NAME, FIRMWARE_UPDATE_VERIFICATION_FILE_MODE_WRITE);
```
[the remark this answers is not in the document]
Comment (srdjastankovic) [reply examples/full_feature_set/firmware_update_implementation.c:153]: tell me which mode to use?

**Review by LazarBozic94 — APPROVED** [review]

**Pull request merged**